### PR TITLE
migrates task to tasks-svc

### LINKED ISSUE #767: Finish migration of task-svc to tasks-svc
### Describe the chore

Take the missing subservices (ward, task-templates, ...) from task-svc and move them to tasks-svc

Furthermore the endpoint `patient/GetPatientAssignmentByWard` has to be implemented, because the query needs the `room_repo`

### PR DESCRIPTION
**Pull request by PaulKalho**
## Which issues does this pull request close?
closes #767 

stacked on #768

### CODE REVIEW

File: proto/services/tasks_svc/v1/task_template_svc.proto [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+message CreateTaskTemplateResponse {
+  string id = 1;
+}
+
+
+///
+/// GetAllTaskTemplates
+///
+
+message GetAllTaskTemplatesRequest {
```
Comment (FoseFx) [inline proto/services/tasks_svc/v1/task_template_svc.proto:31]: do we not need a per-ward query? not suggesting anything, just wondering

File: proto/services/tasks_svc/v1/task_template_svc.proto [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+message CreateTaskTemplateSubTaskResponse {
+  string id = 1;
+}
+
+///
+/// GetAllTaskTemplatesByCreator
+///
+
+message GetAllTaskTemplatesByCreatorRequest {
+  string created_by = 1;
```
Comment (FoseFx) [inline proto/services/tasks_svc/v1/task_template_svc.proto:124]:
```suggestion
  string created_by = 1; // @gotags: validate:"required"
```

File: proto/services/tasks_svc/v1/task_template_svc.proto [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+  optional bool private_only = 2;
```
Comment (FoseFx) [inline proto/services/tasks_svc/v1/task_template_svc.proto:125]: why a tristate here?

File: proto/services/tasks_svc/v1/task_template_svc.proto [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    }
+  }
+}
+
+
+///
+/// GetAllTaskTemplatesByWard
+///
+
+message GetAllTaskTemplatesByWardRequest {
```
Comment (FoseFx) [inline proto/services/tasks_svc/v1/task_template_svc.proto:151]: again, merge this with the ByCreator query into one with optionals

File: proto/services/tasks_svc/v1/ward_svc.proto [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    string id = 1;
+    string name = 2;
+  }
+}
+
+//
+// GetRecentWards
+//
+
+message GetRecentWardsRequest {}
```
Comment (FoseFx) [inline proto/services/tasks_svc/v1/ward_svc.proto:51]:
```suggestion
message GetRecentWardsRequest {}

```

File: proto/services/tasks_svc/v1/ward_svc.proto [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+message DeleteWardResponse {}
+
+
+message GetWardDetailsRequest {
+  string id = 1; // @gotags: validate:"uuid4"
+}
+
+message GetWardDetailsResponse {
+  string id = 1; // @gotags: validate:"uuid4"
```
Comment (FoseFx) [inline proto/services/tasks_svc/v1/ward_svc.proto:92]:
```suggestion
  string id = 1;
```

File: proto/services/tasks_svc/v1/ward_svc.proto [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+
+service WardService {
+  rpc CreateWard (CreateWardRequest) returns (CreateWardResponse) {}
+  rpc GetWard (GetWardRequest) returns (GetWardResponse) {}
+  rpc GetWards(GetWardsRequest) returns (GetWardsResponse) {}
+  rpc GetRecentWards(GetRecentWardsRequest) returns (GetRecentWardsResponse) {}
+  rpc UpdateWard (UpdateWardRequest) returns (UpdateWardResponse) {}
+  rpc DeleteWard (DeleteWardRequest) returns (DeleteWardResponse) {}
+  rpc GetWardOverviews (GetWardOverviewsRequest) returns (GetWardOverviewsResponse) {}
+  rpc GetWardDetails (GetWardDetailsRequest) returns (GetWardDetailsResponse) {}
```
Comment (FoseFx) [inline proto/services/tasks_svc/v1/ward_svc.proto:152]: Can we make sure the order of rpcs here is the same as above (in all services)?

File: proto/services/tasks_svc/v1/task_template_svc.proto (hunk 1 above)
Comment (PaulKalho) [reply proto/services/tasks_svc/v1/task_template_svc.proto:31]: Obviously not, same protos as in task-svc

File: proto/services/tasks_svc/v1/room_svc.proto [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+    }
+  }
+}
+
+//
+// GetRoomsByWard
+//
+
+message GetRoomsByWardRequest {
+  string ward_id = 1; // @gotags: validate:"uuid4"
```
[the remark this answers is not in the document]
Comment (PaulKalho) [reply proto/services/tasks_svc/v1/room_svc.proto:67]: will be noted

File: proto/services/tasks_svc/v1/task_template_svc.proto (hunk 3 above)
Comment (PaulKalho) [reply proto/services/tasks_svc/v1/task_template_svc.proto:125]: will be noted

File: proto/services/tasks_svc/v1/task_template_svc.proto (hunk 4 above)
Comment (PaulKalho) [reply proto/services/tasks_svc/v1/task_template_svc.proto:151]: will be noted

File: proto/services/tasks_svc/v1/room_svc.proto (hunk 8 above)
[the remark this answers is not in the document]
Comment (PaulKalho) [reply proto/services/tasks_svc/v1/room_svc.proto:67]: #807

File: proto/services/tasks_svc/v1/task_template_svc.proto (hunk 3 above)
Comment (PaulKalho) [reply proto/services/tasks_svc/v1/task_template_svc.proto:125]: #807

File: proto/services/tasks_svc/v1/task_template_svc.proto (hunk 4 above)
Comment (PaulKalho) [reply proto/services/tasks_svc/v1/task_template_svc.proto:151]: #807

**Review by FoseFx — APPROVED** [review]

**Pull request merged**